File: server/src/main/java/com/ivantrykosh/app/budgettracker/server/dtos/TransactionDto.java
package com.ivantrykosh.app.budgettracker.server.dtos;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.Date;

/**
 * Transaction dto
 */
@NoArgsConstructor
@Getter
@Setter
@ToString
public class TransactionDto {
    private Long transactionId; // Transaction ID

    private Long accountId; // ID of account to which transaction belongs

    private String category; // Category of transaction

    private Double value; // Value of transaction

    private Date date; // Date of transaction

    private String toFromWhom; // Transaction from whom or to whom

    private String note; // Transaction note
}

File: client/app/src/main/java/com/ivantrykosh/app/budgettracker/client/data/remote/dto/TransactionDto.kt
package com.ivantrykosh.app.budgettracker.client.data.remote.dto

import java.util.Date

/**
 * Data class for transaction
 */
data class TransactionDto(
    val transactionId: Long,
    val accountId: Long,
    val category: String,
    val value: Double,
    val date: Date,
    val toFromWhom: String,
    val note: String
)
